platform/engineering/design-docs/2020-06-02-Sentry10.md

```diff
@@ -0,0 +1,142 @@
+
+
+## Specifics
+### Detailed Design
+
+The [`getsentry/onpremise`](https://github.com/getsentry/onpremise) repo utilizes 18 docker images to run Sentry 10:
+
+```
+sentry_onpremise_clickhouse_1               /entrypoint.sh                   Up      8123/tcp, 9000/tcp, 9009/tcp
+sentry_onpremise_cron_1                     /bin/sh -c exec /docker-en ...   Up      9000/tcp                    
+sentry_onpremise_kafka_1                    /etc/confluent/docker/run        Up      9092/tcp                    
+sentry_onpremise_memcached_1                docker-entrypoint.sh memcached   Up      11211/tcp                   
+sentry_onpremise_post-process-forwarder_1   /bin/sh -c exec /docker-en ...   Up      9000/tcp                    
+sentry_onpremise_postgres_1                 docker-entrypoint.sh postgres    Up      5432/tcp                    
+sentry_onpremise_redis_1                    docker-entrypoint.sh redis ...   Up      6379/tcp                    
+sentry_onpremise_sentry-cleanup_1           /entrypoint.sh 0 0 * * * g ...   Up      9000/tcp                    
+sentry_onpremise_smtp_1                     docker-entrypoint.sh exim  ...   Up      25/tcp                      
+sentry_onpremise_snuba-api_1                ./docker_entrypoint.sh api       Up      1218/tcp                    
+sentry_onpremise_snuba-cleanup_1            /entrypoint.sh */5 * * * * ...   Up      1218/tcp                    
+sentry_onpremise_snuba-consumer_1           ./docker_entrypoint.sh con ...   Up      1218/tcp                    
+sentry_onpremise_snuba-replacer_1           ./docker_entrypoint.sh rep ...   Up      1218/tcp                    
+sentry_onpremise_symbolicator-cleanup_1     /entrypoint.sh 55 23 * * * ...   Up      3021/tcp                    
+sentry_onpremise_symbolicator_1             /bin/bash /docker-entrypoi ...   Up      3021/tcp                    
+sentry_onpremise_web_1                      /bin/sh -c exec /docker-en ...   Up      0.0.0.0:9000->9000/tcp      
+sentry_onpremise_worker_1                   /bin/sh -c exec /docker-en ...   Up      9000/tcp                    
+sentry_onpremise_zookeeper_1                /etc/confluent/docker/run        Up      2181/tcp, 2888/tcp, 3888/tcp
+```
+
+For comparison sake, our Sentry 9 uses 3 docker containers:
+
+- web
+- worker
+- cron
+
+
+### Code Location
+All code will live in the [`devops` repo](https://github.com/department-of-veterans-affairs/devops)
+
+### Testing Plan
+We will stand up Sentry 10 in parallel with our existing Sentry 9.1.2. We will then configure the `Settings.sentry.dsn` of `dev` and/or `staging` to use the Sentry 10 deployment.
+
+### Logging
+N/A
+
+### Debugging
+N/A - I think?
+
+### Caveats
+_To be determined._
+
+### Security Concerns
+No additional security concerns to the ones we have with our existing Sentry.
+
+### Privacy Concerns
+No additional privacy concerns to the ones we have with our existing Sentry.
+
+### Open Questions and Risks
```

Comment:
Another item here is that we're relying upon RDS' snapshots feature to grab backups during deploys/upgrades and just automatically ever day. We'll need to either mitigate that or just accept that the error reports could be wiped out if something goes wrong.

Reply:
I don't even know if a database snapshot is sufficient anymore given the number of extra services that this upgrade adds to the stack.

Reply:
i see pg container in the list, perhaps stop/snapshot ebs/start?  I assume there's no "quiesce service" button